packages/fast-components-react-msft/src/button/button.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import { get } from "lodash-es";
+import { Foundation, HandledProps } from "@microsoft/fast-components-react-base";
+import { IButtonHandledProps, IButtonManagedClasses, IButtonUnhandledProps } from "./button.props";
+import { IManagedClasses, IMSFTButtonClassNameContract } from "@microsoft/fast-components-class-name-contracts-msft";
+import { default as BaseButton } from "@microsoft/fast-components-react-base/src/button";
+
+/* tslint:disable-next-line */
+class Button extends Foundation<IButtonHandledProps & IManagedClasses<IMSFTButtonClassNameContract>,  React.AllHTMLAttributes<HTMLElement>, {}> {
+    protected handledProps: HandledProps<IButtonHandledProps & IManagedClasses<IMSFTButtonClassNameContract>> = {
+        children: void 0,
+        disabled: void 0,
+        href: void 0,
+        justified: void 0,
+        lightweight: void 0,
+        managedClasses: void 0,
+        outline: void 0,
+        primary: void 0
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLButtonElement | HTMLAnchorElement> {
+        return (
+            <BaseButton
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
```

Comment:
Curious on how this may work. Won't managed classes end up being applied twice here? Once in classNames and once in the prop?

Reply:
I see what you are saying but, A) tslint complains immediately if I removed the `managedClasses={this.props.managedClasses}` and B) the output looks as expected: `<button class="button-0-1-66 button_outline-0-1-68">Outline button</button>`